hircluster.h

```diff
@@ -64,61 +68,75 @@ typedef struct cluster_slot {
 } cluster_slot;
 
 typedef struct copen_slot {
-    uint32_t slot_num; /* slot number */
-    int migrate;       /* migrating or importing? */
-    sds remote_name; /* name for the node that this slot migrating to/importing
-                        from */
+    uint32_t slot_num;  /* slot number */
+    int migrate;        /* migrating or importing? */
+    sds remote_name;    /* name of node this slot migrating to/importing from */
     cluster_node *node; /* master that this slot belong to */
 } copen_slot;
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-/* Context for a connection to Redis cluster */
+/* Context for accessing a Redis Cluster */
 typedef struct redisClusterContext {
     int err;          /* Error flags, 0 when there is no error */
     char errstr[128]; /* String representation of error when applicable */
 
-    int flags;
+    /* Configurations */
+    int flags;                                  /* Configuration flags */
+    struct timeval *connect_timeout;            /* TCP connect timeout */
+    struct timeval *command_timeout;            /* Receive and send timeout */
+    int max_redirect_count;                     /* Allowed retry attempts */
+    char password[CONFIG_AUTHPASS_MAX_LEN + 1]; /* Include a null terminator */
```

Comment:
Added fields? Is this new functionality or are they moved from somewhere else?

Reply:
there is no new func. here, I guess the diff got crazy after adding comments

Reply:
Ah, the cluster context and async cluster context were mixed in the diff.